src/java.base/share/classes/javax/crypto/spec/HPKEParameterSpec.java

```diff
@@ -0,0 +1,480 @@
+ * the pre-shared key for authentication and its identifier must be provided
+ * using the {@link #psk(SecretKey, byte[])} method by both sides.
+ * <li>
+ * In HPKE, a shared secret is negotiated during the KEM step and a key
+ * encapsulation message must be transmitted from the sender to the recipient
+ * so that the recipient can recover this shared secret. On the sender side,
+ * the key encapsulation message can be retrieved using the {@link Cipher#getIV()}
+ * method after the cipher is initialized. On the recipient side, the key
+ * encapsulation message can be provided using the {@link #encapsulation(byte[])}
+ * method.
+ * </ul>
+ * For successful interoperability, both sides need to supply identical
+ * {@code info}, {@code psk}, and {@code psk_id} or matching authentication
+ * keys if provided. For details about HPKE modes, refer to
+ * <a href="https://www.rfc-editor.org/rfc/rfc9180.html#section-5">Section 5</a>
+ * of RFC 9180.
+ * <p>
+ * If the sender cipher is initialized without parameters, it assumes a
+ * default parameters object is used, which is equivalent to
+ * {@code HPKEParameterSpec.of()}. The recipient cipher can also be initialized
+ * with a {@code new IvParameterSpec(encap)} object, which is equivalent to
+ * {@code HPKEParameterSpec.of().encapsulation(encap)}. In either case, the
+ * cipher always works in {@code mode_base} mode with an empty {@code info}.
+ * If the recipient side is initialized without any parameters, an
+ * {@code InvalidKeyException} will be thrown.
+ * <p>
+ * At HPKE cipher initialization, if no HPKE implementation supports the provided
+ * key type, an {@code InvalidKeyException} should be thrown. If an
+ * {@code HPKEParameterSpec} is provided but it is not supported by any HPKE
+ * implementation, an {@code InvalidAlgorithmParameterException} will be thrown.
+ * For example:
+ * <ul>
+ * <li> The algorithm identifiers do not match the provided key type.
+ * <li> An attempt to use {@code authKey(key)} is made with an incompatible key.
+ * <li> An attempt to use {@code authKey(key)} is made but the selected KEM
+ *      does not support authentication.
+ * </ul>
+ * Example:
+ * {@snippet lang = java:
+ * // Key pair generation
+ * KeyPairGenerator g = KeyPairGenerator.getInstance("X25519");
+ * KeyPair kp = g.generateKeyPair();
+ *
+ * // The HPKE sender side is initialized with the recipient's public key
+ * Cipher sender = Cipher.getInstance("HPKE");
+ * HPKEParameterSpec ps = HPKEParameterSpec.of()
+ *         .info("this_info".getBytes(StandardCharsets.UTF_8));
+ * sender.init(Cipher.ENCRYPT_MODE, kp.getPublic(), ps);
+ *
+ * // Retrieve the key encapsulation message (the KEM output) from the sender
+ * byte[] kemEncap = sender.getIV();
+ *
+ * // The HPKE recipient side is initialized with its own private key
+ * // and the key encapsulation message from the sender
+ * Cipher recipient = Cipher.getInstance("HPKE");
+ * HPKEParameterSpec pr = HPKEParameterSpec.of()
```

Comment:
I think it would be useful to give an example with some of the constants.

Reply:
I've just pushed a new commit that includes them in the example (the screenshot in the description is also updated). While I haven't used them in `of`, they are compared to values retrieved through `getParameters`.